### PR TITLE
Test coverage 100% + minor improvements

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

• Remove constructor arguments from Factory::create()
• If not instantiable `ContainerInterface`, then use `Factory` for create `Injector` in `CallableDefinition`
• Remove `InvalidDefinition`
• Remove `DefinitionExtractor::fromCallable()`
• Fix and improve exception message in `ClassDefinition`
• Improve exception message in `DefinitionValidator`
• Improve PHPUnit configuration
• Refactoring `ArrayDefinition` for test coverage
• Test coverage 100%

### CODE REVIEW

File: src/Definition/CallableDefinition.php [hunk 1]
```diff
@@ -33,8 +33,12 @@ public function resolve(ContainerInterface $container)
...
         $callable = $this->prepareCallable($this->method, $container);
 
-        /** @psalm-suppress MixedMethodCall */
-        return $container->get(Injector::class)->invoke($callable);
+        /** @var ContainerInterface $injectorContainer */
+        $injectorContainer = $container->has(ContainerInterface::class)
+            ? $container->get(ContainerInterface::class)
+            : $container;
+
+        return (new Injector($injectorContainer))->invoke($callable);
```
[the remark this answers is not in the document]
Comment (samdark) [reply src/Definition/CallableDefinition.php:41]: That would eliminate an ability to set your own Injector instance in container.

[the remark this answers is not in the document]
Comment (vjik) [reply src/Definition/CallableDefinition.php]: Added default definitions. Tests sucessfull in DI and demo.

File: src/Factory.php [hunk 2]
```diff
@@ -170,4 +172,15 @@ public function has($id): bool
...
     }
+
+    private function setDefaultDefinitions(): void
+    {
+        /** @var ContainerInterface */
+        $container = $this->container ?? $this;
+
+        $this->setMultiple([
+            ContainerInterface::class => $container,
+            Injector::class => new Injector($container),
```
Comment (samdark) [inline src/Factory.php:183]: Shouldn't it try to get injector instance from container first?

Comment (vjik) [reply src/Factory.php]: Removed default definition for `Injector`. In `CallableDefinition` injector is gets out from container:

`$container->get(Injector::class)->invoke($callable);`

**Review by samdark — APPROVED** [review]

**Pull request merged**